quiz/quiz.go

```
@@ -0,0 +1,182 @@
+/*
+	Quiz defines a quiz and its questions and provides methods for starting and processing quizzes as well as reading in
+a Quiz from file.
+*/
+package quiz
+
+import (
+	"bufio"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/tabwriter"
+	"time"
+)
+
+/*
+	This Quiz struct is a collection of up to 100 Questions, the name of the file containing the quiz, the number of
+questions in the quiz and the number of questions the user got correct.
+*/
+type Quiz struct {
+	Questions        [100]Question
+	Filename         string
+	TotalQuestions   int
+	CorrectQuestions int
+}
+
+/*
+	Creates a new Quiz struct based on the information in the file at the given file name
+*/
+func NewQuiz(filename string) (quiz Quiz, err error) {
+	fmt.Println("\nGetting quiz using file:\t", filename)
+
+	// open the file and save question to a Quiz struct
+	var questionCount = 0
+	csvFile, _ := os.Open(filename)
```

Probably shouldn't ignore the error here